ChangeLog.md

```diff
@@ -31,6 +31,17 @@ See docs/process.md for more on how version tagging works.
   overflow will trap rather corrupting global data first).  This should not
   be a user-visible change (unless your program does something very odd such
   depending on the specific location of stack data in memory). (#18154)
+- Default value for `STACK_SIZE` (the size of the wasm shadow stack) was reduced
+  from 5Mb to 64Mb.  Project that use more that 64Kb of stack will now have to
+  start specifying `-sSTACK_SIZE` at link time.  To aid in debugging, as of
+  #18154, we now also place the stack first in memory in debug builds so that
+  overflows will be immediately detected, and result in runtime errors.
+  This change brings emscripten into line with wasm-ld and wasi-sdk defaults.
+  In general, WebAssembly stack usage should be lower than on other platforms
+  since a lot of state normally stores on the stack is hidden from the runtime
+  and doesn't use the shadow stack at all.
```

Comment:
Calling the Emscripten/LLVM stack "shadow stack" might read a bit confusing, I don't think the term shadow stack has been used to denote this stack before. LLVM has had a few different Shadow Stack features ( https://clang.llvm.org/docs/ShadowCallStack.html , https://www.phoronix.com/news/LLVM-Drops-ShadowCallStack-x64 ). In the context of Emscripten, in previous conversations that I have been part of, "shadow stack" has generally been used to refer to a program manually managed stack, e.g. for GC purposes.

To LLVM, the stack here is the "native" stack, i.e. the same general stack it uses on all other platforms as well

I've favored the term "Emscripten/LLVM spillover stack" in conversations where distinguishing between the internal Wasm VM stack is needed, since this stack is something that Wasm only needs to do when spilling arrays/structs that need array indexing or otherwise would be ridiculous to model in locals. Would that be a good term to use here?

Reply:
I've personally always used the term "shadow stack" for this.    Perhaps I will bring it up in our next tools meet and we can see if we want to continue to refer to it as such, or if we want to come up with a new name. 

Its not surprising the using a shadow stack for GC purposes used that same name I think since its basically doing the same thing:  Spilling stuff that would otherwise be hidden in the inaccessible wasm stack.  In the GC case the purpose is so that the roots can be seen, and in the C/C++ purpose is to give stack variables and address in linear memory, but they are kind of doing the same thing, right?

@dschuff @kripken do you have preference here, or a suggestion for a different name?

Reply:
LLVM doesn't kind of "know about" the Wasm VM stack, but to LLVM "the stack" is the same native program stack that it defines for all platforms. To me "shadow stack" always meant a stack that is an extra copy of the standard LLVM stack abstraction, typically optionally managed by application developer as an extra feature. So flipping things around to call the hidden Wasm VM stack the "normal" stack and switching to call the LLVM stack a "shadow stack" feels upside down.

Anyhow, not particularly important, either way works for me.

Reply:
I personally like "user stack" because it is a stack in "userspace". That seems least confusing, or at least I can't immediately think of what it could be confused with :smile:

Reply:
We chatted about this in the tools meeting and it does seem that within out team we have been (for better or worse) using the term shadow stack for a while now.    I think it might be technically wrong, so I will try to re-word this entry to avoid it.